frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardPage.unit.spec.tsx

```diff
@@ -98,6 +114,44 @@ describe("PublicOrEmbeddedDashboardPage", () => {
 
     expect(screen.getByText("There's nothing here, yet.")).toBeInTheDocument();
   });
+
+  describe("downloads flag", () => {
+    it("should show the 'Export as PDF' button even when titled=false and there's one tab", async () => {
+      await setup({ hash: "titled=false", numberOfTabs: 1 });
+
+      expect(screen.getByText("Export as PDF")).toBeInTheDocument();
+    });
+
+    it('should not show the "Export as PDF" button when downloads are disabled', async () => {
+      await setup({ hash: "downloads=false", numberOfTabs: 1 });
+
+      expect(screen.queryByText("Export as PDF")).not.toBeInTheDocument();
+    });
+
+    it("should allow downloading the dashcards results when downloads are enabled", async () => {
+      await setup({ numberOfTabs: 1, hash: "downloads=true" });
+
+      await userEvent.click(getIcon("ellipsis"));
+
+      expect(screen.getByText("Download results")).toBeInTheDocument();
+    });
+
+    it("should not allow downloading the dashcards results when downloads are disabled", async () => {
+      await setup({ numberOfTabs: 1, hash: "downloads=false" });
+
+      // in this case the dashcard menu would be empty so it's not rendered at all
+      expect(queryIcon("ellipsis")).not.toBeInTheDocument();
+    });
+
+    it("should use the container used for pdf exports", async () => {
+      const { container } = await setup({ numberOfTabs: 1 });
+
```

Comment:
this is not a *great* test, but it's the quickest way to check if we totally broke the feature.
I'll try and so some other proper test in the e2e, possibly testing if we export a non-empty pdf